os/drivers/audio/alc5658.c

```diff
@@ -644,52 +507,71 @@ static int alc5658_configure(FAR struct audio_lowerhalf_s *dev, FAR const struct
 #endif							/* CONFIG_AUDIO_EXCLUDE_VOLUME */
 
 #ifndef CONFIG_AUDIO_EXCLUDE_TONE
-		case AUDIO_FU_BASS: {
-			/* Set the bass.  The percentage level (0-100) is in the
-			 * ac_controls.b[0] parameter.
-			 */
+		case AUDIO_FU_BASS:
+		{	/* Set the bass.  The percentage level (0-100) is in the
+			 * ac_controls.b[0] parameter. */
 
 			uint8_t bass = caps->ac_controls.b[0];
 			audvdbg("    Bass: %d\n", bass);
-
-			if (bass <= 100) {
+			if (bass <= 100)
 				alc5658_setbass(priv, bass);
-			} else {
-				ret = -EDOM;
-			}
 		}
 		break;
-
-		case AUDIO_FU_TREBLE: {
-			/* Set the treble.  The percentage level (0-100) is in the
-			 * ac_controls.b[0] parameter.
-			 */
+		case AUDIO_FU_TREBLE:
+		{	/* Set the treble.  The percentage level (0-100) is in the
+			 * ac_controls.b[0] parameter. */
 
 			uint8_t treble = caps->ac_controls.b[0];
 			audvdbg("    Treble: %d\n", treble);
-
-			if (treble <= 100) {
+			if (treble <= 100)
 				alc5658_settreble(priv, treble);
-			} else {
-				ret = -EDOM;
-			}
 		}
 		break;
 #endif							/* CONFIG_AUDIO_EXCLUDE_TONE */
-
 		default:
 			auddbg("    ERROR: Unrecognized feature unit\n");
-			ret = -ENOTTY;
 			break;
 		}
-		break;
+		break; /* Break for inner switch case */
+	case AUDIO_TYPE_INPUT: {
```

Comment:
AUTIO_TYPE_INPUT is same as AUDIO_TYPE_OUTPUT..
also we can remove 
#if 0
~~
#endif

Reply:
I fixed.